src/nvim/eval.c

```diff
@@ -10724,15 +10734,13 @@ static void f_jobsend(typval_T *argvars, typval_T *rettv)
     return;
   }
 
-  Job *job = job_find(argvars[0].vval.v_number);
-
-  if (!is_user_job(job)) {
-    // Invalid job id
+  TerminalJobData *data = pmap_get(uint64_t)(jobs, argvars[0].vval.v_number);
+  if (!data) {
     EMSG(_(e_invjob));
     return;
   }
 
-  if (((TerminalJobData *)job_data(job))->stdin_closed) {
+  if (data->stdin_closed) {
```

Comment:
Do we still need `stdin_closed` in `TerminalJobData`?
We could now do something like this:
```c
  Process *proc = (Process *)&data->proc;
  if (!proc->in || proc->in->closed) {
```